app/scrappers/ScrapperSupervisor.scala

```diff
@@ -19,13 +23,58 @@ class ScrapperSupervisor @Inject()
     implicit ec: ExecutionContext,
     ws: WSClient,
     mongo: Mongo,
-    lifecycle: ApplicationLifecycle
+    lifecycle: ApplicationLifecycle,
+    appConfig: Configuration
   ) extends Actor {
 
   override val supervisorStrategy = (new StoppingSupervisorStrategy).create()
   val logger = Logger(this.getClass)
 
-  override def preStart() {
+  /* Reads the `scrappers.toml` configuration file and start the scrappers defined.
+   *
+   * Additional documentation about defining scrappers can be
+   * found in the aforementioned file.
+   */
+  def startScrappersFromConfig(configPath: String = appConfig.get[String]("soxx.scrappers.configFile")) {
+    Try {
+      val cfgFile = scala.io.Source.fromFile(configPath)
+      try cfgFile.mkString finally cfgFile.close
+    } match {
+      case Success(configStr) =>
```

Comment:
Can't we first handle erroneous cases and then unwrap value?
To reduce code folding level.